### DISCUSSION TITLE
Menampilkan faq.data dari Page (parent page) ke semua Child-nya

### DISCUSSION THREAD

**Discussion by bungfrangki** [mecha-cms/mecha#275] [Q&A] [answered]
Entah istilah di judul itu benar atau tidak. Kurang lebih seperti ini ilustrasinya:

```
.\
└── lot\
    └── page\
        ├── parent-page\
        │   └── faq.data
        │   └── page-child-1
        │   └── page-child-2
        │   └── page-child-3
        │   └── page-child-1.page
        │   └── page-child-2.page
        │   └── page-child-3.page
        └── parent-page.page
```

isi konten `faq.data` berupa baris kode seperti ini:

```html
<section class='faq'>
<h2>FAQ</h2>
<h3>Question 1</h3>
<p>Answer 1</p>	
<h3>Question 2</h3>
<p>Answer 2</p>	
<h3>Question 3</h3>
<p>Answer 3</p>	
</section>
```

Nah, saya ingin menampilkan `faq.data` tersebut ke Parent Page dan ke semua Child-nya.

Kurang lebih seperti ini yang saya terapkan ke `page.php`

```php
<?php if ($site->has('part')): ?>
   <div>
	<?= $page->faq; ?>
   </div>
<?php elseif ($site->has('parent')):?>
   <div>
	<?= $page->parent->faq; ?>
   </div>
<?php else: ?>
   <div>
	<?= $page->faq; ?>
   </div>
 <?php endif; ?>
```

mungkin ada tanggapan dan tambahan, disilahkan! thx

**Comment by taufik-nurrohman** [comment, marked as answer]
Kalau isinya JSON, bisa diiterasikan, sehingga markup HTML tidak perlu ditulis di dalam data:

~~~ json
{
  "title": "FAQ",
  "list": [
    {
      "q": "Question 1",
      "a": "Answer 1"
    },
    {
      "q": "Question 2",
      "a": "Answer 2"
    }
  ]
}
~~~

Terus di layout bisa begini:

~~~ php
<h2>
  <?= $page->faq['title'] ?? 'Default'; ?>
</h2>
<?php if (!empty($page->faq['list'])): ?>
  <?php foreach ($page->faq['list'] as $v): ?>
    <h3>
      <?= $v['q'] ?? 'Default'; ?>
    </h3>
    <p>
      <?= $v['a'] ?? 'Default'; ?>
    </p>
  <?php endforeach; ?>
<?php endif; ?>
~~~

**Reply by bungfrangki** [reply]
wah, ini paling mantap (y)
thx mas

**Comment by bungfrangki** [comment]
> Kalau isinya JSON, bisa diiterasikan, sehingga markup HTML tidak perlu ditulis di dalam data:
> 
> ```json
> {
>   "title": "FAQ",
>   "list": [
>     {
>       "q": "Question 1",
>       "a": "Answer 1"
>     },
>     {
>       "q": "Question 2",
>       "a": "Answer 2"
>     }
>   ]
> }
> ```
>

**Reply by bungfrangki** [reply]
mas, JSON ini kalau dipanggil menggunakan jQuery bisa gak mas?

**Reply by taufik-nurrohman** [reply]
Di-dump aja ke variabel JavaScript:

~~~ html
<script>const faq = <?= To::JSON($page->faq); ?>;</script>
~~~
